### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by GraceHtet**
- Create a class Classroom 

- Create the has-many/belongs-to relationship between Classroom and Student:

> Create the has-many side (a classroom has many students).

> Create the belongs-to side (a student belongs to a classroom).

- Create a class Book 

- Create a class Rental

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental:

> Create the has-many side of Book and Rental (a book has many rentals).

> Create the belongs-to side of Rental and Book (a rental belongs to a book).

> Create the has-many side of Person and Rental (a person has many rentals).

> Create the belongs-to side of Rental and Person (a rental belongs to a person).

> Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: lib/book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rents
```
Comment (rloterh) [inline lib/book.rb]: - [ ] `rental` (in your case _rents_) should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ it. Kindly ensure to improve upon this aspect.

File: lib/book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Book
```
Comment (rloterh) [inline lib/book.rb:1]: - [ ] A _rental_ can have many _books_, and _books_ can be subjected to many _rentals_. In order for us to establish a _many-to-many_ association between _rentals_ and _books_ (let's bear in mind that the association between _person_ and _rentals_ is also _many-to-many_), let's define a method for adding rentals within the book object similar to the example below:

```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .

  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: lib/rental.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
```
Comment (rloterh) [inline lib/rental.rb]: - [ ] Similarly, `book` and `person` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ to it. Kindly ensure to improve upon this aspect.

File: lib/person.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 require_relative '../nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rents
```
Comment (rloterh) [inline lib/person.rb]: - [ ] Again, `rental` (in your case _rents_) should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _modify (set)_ it. Kindly ensure to improve upon this aspect.

File: lib/person.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 require_relative '../nameable'
 
 class Person < Nameable
```
Comment (rloterh) [inline lib/person.rb:3]: - [ ] Let's ensure a method for adding _rentals_ is present, so as to establish a a _rental_ `belong-to` _person_ relationship, like the code example below:

```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def add_rental(book, date)
    Rental.new(date, book, self)
  end

  .
  .
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Grace, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁  🌴 🚦 🚗 🏫  🏡   _🏃 

## Highlights 👍  
- Good job so far establishing associations for book library entities! 
- ReadMe looks good!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- Kindly endeavor to remove links from the table contents from your ReadMe which do not have related sections/contents.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @GraceHtet,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**